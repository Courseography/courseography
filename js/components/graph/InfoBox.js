import React from "react"
import PropTypes from "prop-types"

export default class InfoBox extends React.Component {
  render() {
    var className = this.props.showInfoBox
      ? "tooltip-group-display"
      : "tooltip-group-hidden"

    var rectAttrs = {
      id: this.props.nodeId + "-tooltip" + "-rect",
      x: this.props.xPos,
      y: this.props.yPos,
      rx: "4",
      ry: "4",
      fill: "white",
      stroke: "black",
      strokeWidth: "2",
      width: "60",
      height: "30",
    }

    var textAttrs = {
      id: this.props.nodeId + "-tooltip" + "-text",
      x: this.props.xPos + 60 / 2 - 18,
      y: this.props.yPos + 30 / 2 + 6,
    }

    return (
      <g
        id="infoBox"
        className={className}
        onClick={this.props.onClick}
        onMouseEnter={this.props.onMouseEnter}
        onMouseLeave={this.props.onMouseLeave}
      >
        <rect {...rectAttrs} />
        <text {...textAttrs}>Info</text>
      </g>
    )
  }
}

InfoBox.propTypes = {
  showInfoBox: PropTypes.bool,
  nodeId: PropTypes.string,
  xPos: PropTypes.number,
  yPos: PropTypes.number,
  onClick: PropTypes.func,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func,
}
