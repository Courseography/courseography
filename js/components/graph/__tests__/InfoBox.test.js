import React from "react"
import { shallow } from "enzyme"
import InfoBox from "../InfoBox"
import TestGraph from "./TestGraph"
import { fireEvent } from "@testing-library/react"

describe("InfoBox", () => {
  it("should match shallow snapshot", () => {
    const infoBoxProps = {
      onClick: jest.fn(),
      onMouseDown: jest.fn(),
      onMouseLeave: jest.fn(),
      nodeId: "",
      showInfoBox: false,
      xPos: 0,
      yPos: 0,
    }
    const component = shallow(<InfoBox {...infoBoxProps} />)
    expect(component).toMatchSnapshot()
  })

  it("should appear when hovering over a course", async () => {
    const graph = await TestGraph.build()
    const aaa100 = graph.getByTestId("aaa100")

    const infoBox = graph.getNodeByText("Info")
    expect(infoBox.classList.contains("tooltip-group-hidden")).toBe(true)
    fireEvent.mouseOver(aaa100)
    expect(infoBox.classList.contains("tooltip-group-display")).toBe(true)
  })

  it("should disappear a second after the the cursor isn't hovered over the course", async done => {
    const graph = await TestGraph.build()
    const aaa100 = graph.getByTestId("aaa100")

    fireEvent.mouseOver(aaa100)
    const infoBox = graph.getNodeByText("Info")
    expect(infoBox.classList.contains("tooltip-group-display")).toBe(true)

    fireEvent.mouseOut(aaa100)

    setTimeout(() => {
      expect(infoBox.classList.contains("tooltip-group-hidden")).toBe(true)
      done()
    }, 1000)
  })
  it("Pressing on the info box should create a new pop up", async done => {
    const graph = await TestGraph.build()
    const aaa100 = graph.getByTestId("aaa100")
    fireEvent.mouseOver(aaa100)
    const infoBox = graph.getNodeByText("Info")
    fireEvent.click(infoBox)

    // wait for fake fetch to finish
    setTimeout(() => {
      // expect description in the modal box to appear
      expect(graph.textExists(/AAA Thinking/)).toBe(true)
      done()
    }, 1000)
  })
})
