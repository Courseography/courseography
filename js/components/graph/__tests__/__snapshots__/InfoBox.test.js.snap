// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox should match shallow snapshot 1`] = `
<g
  className="tooltip-group-hidden"
  id="infoBox"
  onClick={[MockFunction]}
  onMouseLeave={[MockFunction]}
>
  <rect
    fill="white"
    height="30"
    id="-tooltip-rect"
    rx="4"
    ry="4"
    stroke="black"
    strokeWidth="2"
    width="60"
    x={0}
    y={0}
  />
  <text
    id="-tooltip-text"
    x={12}
    y={21}
  >
    Info
  </text>
</g>
`;
